docs/main/docs/components/executor-overview.md

```diff
@@ -4,10 +4,54 @@ sidebar_position: 1
+
+### Circuit Relayer:
+
+An Executor must be able to communicate with the t3rn Circuit. The Executor must be able to submit transactions, this is done via the Circuit Realyer class, which contains all relevant functions. The Executor must also subscribe to state updates. In t3rn, these are emitted via events. The CircuitListener class is responsible for this, forwarding all relevant incoming events.
+
+### Execution Manager:
+
+The execution manager is the class connecting all functionality together. It tracks all incoming Cross Chain Transactions (XTX) throughout the TX lifecycle, processing incoming events and reacting accordingly. At its core, the execution manager tracks the state transitions emitted as an event by the t3rn Circuit, and reacts accordingly.
+For example, if a new bid for a Side Effects (SFX) we have bid on is detected, the execution manager evaluates the SFX, deciding if a counter bid should be submitted.
+
+### Pricing Engine:
+
+To be able to evaluate the profitability of a SFX, the Executor needs to have access to a constant stream of asset prices. The PriceEngine takes care of this, tracking prices for assets that need to be tracked. This enables changing prices to automatically be pushed to all instances that are tracking a certain asset.
+Currently, the PriceEngine is very basic, solely querying prices from CoinGecko. However, it is built with extensibility in mind, allowing new price sources to be added without much configuration.
+
+### Strategy Engine:
+
+The strategy engine is a component used for specifying execution strategies. It allows Executors to define constraints that are then evaluated for incoming XTXs. A strategy is always defined on a per-target basis. The strategy engine has two types of checks it performs:
+
+#### XTX Strategy:
+
+When a new XTX is added, the XTX strategy is evaluated. This is done by iterating through the XTXs SFX and checking that it passes the XTX strategy for its target chain. If any of these evaluations fails, the XTX is ignored and no longer considered for execution.
+
+#### SFX Strategy:
+
+The SFX strategy is used to decide if a certain SFX is deemed profitable. Any SFX that pass the evaluation are deemed favorable, triggering a bid to be generated and submitted. This stage is only reached if the XTX strategy passes beforehand.
+
+### Bidding Engine:
+
+The bidding engine is responsible for generating the bidding amount for a specific SFX. It only receives SFXs that have been deemed profitable by the strategy engine. In its current form, the bidding engine uses [Strategy.minProfitUsd; sfx.profitUsd] as a range and then calculates the bid by returning the amount at a configured quartile.
+
+### Gateway Relayers:
+
+To execute transactions on other blockchains, a custom relayer class is needed. These classes need to be able to execute a certain set of transactions and generate an inclusion proof for said transaction. Currently, a Substrate based version is implemented.
```

Comment:
Would be nice to give a breakdown of fees paid in T0RN for each step

Reply:
Noted and will be fixed in next phase